addons/payment/models/res_company.py

```diff
@@ -50,3 +50,15 @@ def _install_modules(self, module_names):
         modules_sudo = self.env['ir.module.module'].sudo().search([('name', 'in', module_names)])
         STATES = ['installed', 'to install', 'to upgrade']
         modules_sudo.filtered(lambda m: m.state not in STATES).button_immediate_install()
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        companies = super().create(vals_list)
+        providers = self.env['payment.provider'].sudo().search(
+            [('company_id', '=', self.env.user.company_id.id)]
+        )
+        for company in companies:
+            for provider in providers:
+                provider.sudo().copy({'company_id': company.id})
```

Comment:
```suggestion
        providers_sudo = self.env['payment.provider'].sudo().search(
            [('company_id', '=', self.env.user.company_id.id)]
        )
        for company in companies:
            for provider_sudo in providers_sudo:
                provider_sudo.copy({'company_id': company.id})
```
The result of a sudo'ed search already gives sudo'ed record ;).

Reply:
Sure! :+1: